.github/workflows/publish-library.yml

```diff
@@ -25,12 +25,11 @@ jobs:
       - uses: actions/checkout@v3
       - name: Set up Python
         uses: actions/setup-python@v3
-      - name: Release
-        env:
-          FLIT_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+      - name: Build and check package
         run: |
-          pip install tox
-          RELEASE_VERSION=${GITHUB_REF#refs/*/}
-          RELEASE_VERSION=${RELEASE_VERSION#v*}
-          sed -i "s/^version = .*/version = \"${RELEASE_VERSION}\"/" pyproject.toml
-          tox -e publish
+          tox -e build,twinecheck
+      - name: Upload package
+        if: github.event_name == 'release'
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          password: ${{ secrets.PYPI_TOKEN }}
```

Comment:
FYI using a long-living secret is no longer necessary with secretless publishing: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/